ext/spl/php_spl.c

```diff
@@ -625,6 +656,40 @@ PHP_FUNCTION(spl_autoload_functions)
 	}
 } /* }}} */
 
+
+/* {{{ Assign an array of name => path mappings to the autoloader */
+PHP_FUNCTION(autoload_set_classmap)
+{
+	HashTable * ht = NULL;
+	
+	ZEND_PARSE_PARAMETERS_START(1, 1)
+		Z_PARAM_ARRAY_HT(ht)
+	ZEND_PARSE_PARAMETERS_END();
+	
+	if (SPL_G(autoload_classmap)) {
+		zend_throw_error(NULL, "Classmap Autoloader can only be initialized once per request.");
+		RETURN_THROWS();
+	}
+	
+	ALLOC_HASHTABLE(SPL_G(autoload_classmap));
+	zend_hash_init(SPL_G(autoload_classmap), zend_hash_num_elements(ht), NULL, ZVAL_PTR_DTOR, 0);
+	zend_hash_copy(SPL_G(autoload_classmap), ht, (copy_ctor_func_t)zval_add_ref);
+	
+	RETURN_TRUE;
+} /* }}} */
+
+/* {{{ Assign an array of name => path mappings to the autoloader */
+PHP_FUNCTION(autoload_get_classmap)
+{
+	ZEND_PARSE_PARAMETERS_NONE();
+	
+	if (SPL_G(autoload_classmap)) {
+		RETURN_ARR(zend_array_dup(SPL_G(autoload_classmap)));
```

Comment:
No need to copy, do `GC_TRY_ADDREF(SPL_G(autoload_classmap)); RETURN_ARR(SPL_G(autoload_classmap));`